src/odemis/driver/test/lakeshore_test.py

```diff
@@ -34,37 +34,84 @@
 # needing real hardware
 TEST_NOHW = (os.environ.get("TEST_NOHW", "0") != "0")  # Default to Hw testing
 
-# arguments used for the creation of basic components
-CONFIG = {"name": "Lakeshore Test",
-          "role": "temperature-controller",
-          "port": "/dev/ttyUSB0",
-          "sensor_input": "B",
-          "output_channel": 2,
+# arguments used for the creation of basic components to test via serial port connection
+CONFIG_SINGLE_TEMP = {
+    "name": "Lakeshore Test",
+    "role": "temperature-controller",
+    "port": "/dev/ttyUSB0/",
+    # "port": "/dev/fake",
+    "sensor_input": "B",
+    "output_channel": 2,
+}
+
+# arguments used for the creation of basic components to test via IP connection
+CONFIG_MULTI_TEMP = {
+    "name": "Lakeshore Test",
+    "role": "temperature-controller",
+    "port": "192.168.30.214",
+    # "port": "/dev/fake",
+    "sensor_input": {"monolith": "A", "sample": "B"},
+    "output_channel": {"monolith": 1, "sample": 2},
 }
 
 # arguments used for the creation of basic components
 if TEST_NOHW:
-    CONFIG["port"] = "/dev/fake"
+    CONFIG_SINGLE_TEMP["port"] = "/dev/fake"
+    CONFIG_MULTI_TEMP["port"] = "/dev/fake"
+
+
+class LakeshoreBaseTest:
+    """Base class for testing different models of LakeShore temperature controller."""
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.dev.terminate()  # free up socket.
 
 
-class TestLakeshore(unittest.TestCase):
+class TestLakeshoreModel335(LakeshoreBaseTest, unittest.TestCase):
     """
     Tests cases for the Lakeshore 335 Temperature PID
     """
 
+    def setUp(self):
+        self._heating = self.dev.heating.value
+        self._targetTemperature = self.dev.targetTemperature.value
+
+    def tearDown(self):
+        self.dev.targetTemperature.value = self._targetTemperature
+        self.dev.heating.value = self._heating
+
     @classmethod
     def setUpClass(cls):
-        cls.dev = lakeshore.Lakeshore(**CONFIG)
+        cls.dev = lakeshore.Lakeshore(**CONFIG_SINGLE_TEMP)
 
-    @classmethod
-    def tearDownClass(cls):
-        cls.dev.terminate()  # free up socket.
+    def test_device_model(self):
+        """
+        Test device model
+        """
+        manufacturer, model, _, _ = self.dev.GetIdentifier()
+
+        if TEST_NOHW:
+            self.skipTest("Simulator always pretends to be MODEL350.")
+
+        self.assertEqual(model, "MODEL335")
+
+    def test_temperature_poll(self):
+        # Test that the temperature polls.
+        # Designed to work with the simulator - the temperature will not be exactly the same
+        logging.debug("Test stasis temperature polling")
+        temp1 = []
+        temp2 = []
+        for testCtr in range(10):
```

Comment:
In such simple loop, especially where the variable is not used, the convention is to name the variable `i`.